src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -117,7 +117,7 @@ void SmithWatermanAlgorithm::launch(const SMatrix &_substitutionMatrix,
                                     int _minScore,
                                     SmithWatermanSettings::SWResultView _resultView) {
     setValues(_substitutionMatrix, _patternSeq, _searchSeq, _gapOpen, _gapExtension, _minScore, _resultView);
-    if (isValidParams() && calculateMatrixLength()) {
+    if (calculateMatrixLength() && isValidParams()) {
```

Comment:
Is there a reason to swap them?

Reply:
Yep, we cant validate parameter if it wasn't calculated yet.